### ISSUE TITLE
suspicious code in calendarspec.c

### ISSUE THREAD

**Issue by keszybz** [systemd/systemd#2678]
Coverity complains that the return value of `mktime_or_timegm` at line 981 is not checked. It probably should be, because `mktime_or_timegm` does nothing on error. 

Also right below is this code

```
                if (r > 0) {
                        c.tm_mon = 0;
                        c.tm_mday = 1;
                        c.tm_hour = c.tm_min = c.tm_sec = tm_usec = 0;
                }
                if (r < 0 || tm_out_of_bounds(&c, spec->utc))
                        return r;
```

if `tm_out_of_bounds` returns true, function returns >=0, which seems wrong.
I really don't grok this code, so I'm putting this up here in hope somebody can have a look at it.

**Comment by ohsix** [comment]
it might be for the side-effects of mktime/timegm

from mktime man
`The mktime() function modifies the fields of the tm structure as follows: tm_wday and tm_yday are set
       to  values  determined  from the contents of the other fields; if structure members are outside their
       valid interval, they will be normalized (so that, for example, 40 October is changed  into  9  Novem‐
       ber); tm_isdst is set (regardless of its initial value) to a positive value or to 0, respectively, to
       indicate whether DST is or is not in effect at the specified time.  Calling mktime()  also  sets  the
       external variable tzname with information about the current timezone.
`

some other calling sites where the return value isn't use also suggests this might be the case

**Comment by keszybz** [comment]
Right, it's called for side effect. But the sideeffects will not happen if the mktime fails. But I'm not sure if in this case mktime simply cannot fail (which would mean that we can wrap this in `assert_se`), or if we can ignore effects if it fails.

**Comment by poettering** [comment]
Fix waiting in #2688.
